tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,217 @@
+// autoupdate process for tools and agents.
+type AutoUpdateCommand struct {
+	app    *kingpin.Application
+	config *servicecfg.Config
+
+	updateCmd *kingpin.CmdClause
+	getCmd    *kingpin.CmdClause
+	watchCmd  *kingpin.CmdClause
+
+	mode               string
+	toolsTargetVersion string
+	proxy              string
+
+	// stdout allows to switch standard output source for resource command. Used in tests.
+	stdout io.Writer
+}
+
+// Initialize allows AutoUpdateCommand to plug itself into the CLI parser.
+func (c *AutoUpdateCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	c.app = app
+	c.config = config
+	autoUpdateCmd := app.Command("autoupdate", "Teleport auto update commands.")
+
+	clientToolsCmd := autoUpdateCmd.Command("client-tools", "Client tools auto update commands.")
+
+	c.updateCmd = clientToolsCmd.Command("update", "Edit client tools auto update configuration.")
+	c.updateCmd.Flag("set-mode", `Sets the mode to enable or disable tools auto update in cluster.`).EnumVar(&c.mode, "enabled", "disabled")
+	c.updateCmd.Flag("set-target-version", `Defines client tools target version required to be updated.`).StringVar(&c.toolsTargetVersion)
+
+	c.getCmd = clientToolsCmd.Command("get", "Receive tools auto update target version.")
+	c.getCmd.Flag("proxy", `URL of the proxy`).StringVar(&c.proxy)
+
+	c.watchCmd = clientToolsCmd.Command("watch", "Start monitoring auto update target version updates.")
+	c.watchCmd.Flag("proxy", `URL of the proxy`).StringVar(&c.proxy)
+
+	if c.stdout == nil {
+		c.stdout = os.Stdout
+	}
+}
+
+// TryRun takes the CLI command as an argument and executes it.
+func (c *AutoUpdateCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+	switch cmd {
+	case c.updateCmd.FullCommand():
+		err = c.Upsert(ctx, client)
+	case c.getCmd.FullCommand():
+		err = c.Get(ctx, client)
+	case c.watchCmd.FullCommand():
+		err = c.Watch(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// Upsert works with AutoUpdateConfig and AutoUpdateVersion resources to create or update.
+func (c *AutoUpdateCommand) Upsert(ctx context.Context, client *authclient.Client) error {
+	if c.mode != "" {
+		config, err := client.GetAutoUpdateConfig(ctx)
+		if trace.IsNotFound(err) {
+			if config, err = autoupdate.NewAutoUpdateConfig(&autoupdatev1pb.AutoUpdateConfigSpec{}); err != nil {
+				return trace.Wrap(err)
+			}
+		} else if err != nil {
+			return trace.Wrap(err)
+		}
+		if config.Spec.Tools == nil {
+			config.Spec.Tools = &autoupdatev1pb.AutoUpdateConfigSpecTools{}
+		}
+		if config.Spec.Tools.Mode != c.mode {
+			config.Spec.Tools.Mode = c.mode
+			if _, err := client.UpsertAutoUpdateConfig(ctx, config); err != nil {
+				return trace.Wrap(err)
+			}
+			fmt.Fprint(c.stdout, "autoupdate_config has been upserted\n")
+		}
+	}
+
+	version, err := client.GetAutoUpdateVersion(ctx)
```

Comment:
is the version mandatory? If `c.toolsTargetVersion` is empty, this will replace the value